server/src/main/java/org/elasticsearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -711,6 +711,7 @@ public int getNumberOfInFlightFetches() {
      */
     public record CommandsResult(
         RoutingExplanations explanations, // Explanation for the reroute actions
-        ClusterState clusterState         // Resulting cluster state, should be removed in 9
+        ClusterState clusterState         // Resulting cluster state, to be removed when REST compatibility with
+                                          // {@link org.elasticsearch.Version#V_8_6_0} / {@link RestApiVersion#V_8} no longer needed
```

Comment:
Could this be proper Javadoc rather than just a comment. It won't fail the compile here.